config.mk

```diff
@@ -1,12 +1,15 @@
 _VERSION = 0.3.1
-VERSION = $(shell ./generate-version.sh $(_VERSION))
+VERSION  = `./generate-version.sh $(_VERSION)`
 
 # paths
 PREFIX = /usr/local
 MANDIR = $(PREFIX)/share/man
 
-# Default compile flags (overridable by environment)
-CFLAGS ?= -g -Wall -Wextra -Werror -Wno-unused-parameter -Wno-sign-compare -Wno-unused-function -Wno-unused-variable -Wno-unused-result -Wdeclaration-after-statement
+# Compile flags that can be used
+#CFLAGS = -pedantic -Wall -Wextra -Werror -Wno-unused-parameter -Wno-sign-compare -Wno-unused-function -Wno-unused-variable -Wno-unused-result -Wdeclaration-after-statement
 
+XWAYLAND =
+XLIBS =
```

Comment:
Are this lines really necessary?

Reply:
I figured if the user doesn't comment out the lines, the variables should be explicitly defined to expand to nothing so it doesn't affect where they're called in the main `Makefile`.

Reply:
POSIX.1-2017 doesn’t specify what happens with expansions of undefined macros. Actually, we rely on them expanding to nothing all the time though—think `DESTDIR`. That is implemented ubiquitously and standardized in POSIX.1-202x.

These two lines just cause `XWAYLAND` and `XLIBS` not to be imported from the environment. Whether that is useful is up for debate. I would leave them there.